src/routes/notifications/guards/notifications.guard.ts

```diff
@@ -0,0 +1,25 @@
+import { AuthController } from '@/routes/auth/auth.controller';
+import { AuthGuard } from '@/routes/auth/guards/auth.guard';
+import { CanActivate, ExecutionContext, Injectable } from '@nestjs/common';
+import { Request } from 'express';
+
+@Injectable()
+export class NotificationsGuard implements CanActivate {
+  constructor(private readonly authGuard: AuthGuard) {}
+  canActivate(context: ExecutionContext): boolean {
+    const request: Request = context.switchToHttp().getRequest();
+
+    const accessToken: string | undefined =
+      request.cookies[AuthController.ACCESS_TOKEN_COOKIE_NAME];
+
+    /**
+     * If there is no access token, we allow the request to proceed as
+     * users can register for "public" notifications having not signed.
+     */
```

Comment:
As before, what do you think about making this comment more "purpose agnostic"?

Reply:
Updated in da9bf557fe0afb1b1eabea3681f878250e7c90ee.